oarepo_requests/ui/theme/assets/semantic-ui/js/oarepo_requests_ui/common/components/RequestCustomFields.jsx

```diff
@@ -0,0 +1,99 @@
+import React from "react";
+import { Grid, List, Form, Divider } from "semantic-ui-react";
+import { CustomFields } from "react-invenio-forms";
+import {
+  ReadOnlyCustomFields,
+  REQUEST_TYPE,
+  REQUEST_MODAL_TYPE,
+} from "@js/oarepo_requests_common";
+import PropTypes from "prop-types";
+
+export const RequestCustomFields = ({
+  request,
+  customFields,
+  actions,
+  columnWidth,
+}) => {
+  // Only applies to RequestModalContent component:
+  // READ ONLY modal type contains Accept, Decline, and/or Cancel actions OR contains Cancel action only => only ReadOnlyCustomFields are rendered
+  // SUBMIT FORM modal type contains Submit and/or Save, Create, CreateAndSubmit action => Form is rendered
+  const customFieldsType = actions.some(
+    ({ name }) => name === REQUEST_TYPE.ACCEPT || name === REQUEST_TYPE.CANCEL
+  )
+    ? REQUEST_MODAL_TYPE.READ_ONLY
+    : REQUEST_MODAL_TYPE.SUBMIT_FORM;
+
+  const renderSubmitForm =
+    customFieldsType === REQUEST_MODAL_TYPE.SUBMIT_FORM &&
+    customFields?.ui?.length > 0;
+  const customFieldsPaths = customFields?.ui
+    ?.map(({ fields }) => {
+      let paths = [];
+      for (const field of fields) {
+        paths.push(field.field);
+      }
+      return paths;
+    })
+    .flat();
+
+  const renderReadOnlyData =
+    customFieldsType === REQUEST_MODAL_TYPE.READ_ONLY &&
+    Object.keys(request?.payload || {}).some((key) =>
+      customFieldsPaths?.includes(key)
+    );
+
+  return renderSubmitForm || renderReadOnlyData ? (
```

Comment:
Duplicit / unnecessary condition

Reply:
@mirekys it serves to not display an empty row in case none of the conditionals are met, because SUI row even if it does not contain anything just takes up fixed amount of space anyway